sync2/cond_timed_wait.go

```diff
@@ -0,0 +1,48 @@
+package sync2
+
+import (
+	"sync"
+	"sync/atomic"
+	"time"
+)
+
+// Wait for a condition. If the timeout passes, returns 'false'.
+// If the condition is signalled, returns 'true'.
+//
+// The condition's mutex must already be held when calling this function.
+//
+// If reSignal is true, then the condition will be re-signaled on timeout.
+func CondTimedWait(cond *sync.Cond, timeout time.Duration, reSignal bool) bool {
```

Comment:
as discussed via email, we should remove the reSignal argument and unconditionally resignal on timeout